pkg/sql/pgwire/pgwirebase/encoding.go

```diff
@@ -102,9 +109,17 @@ func (b *ReadBuffer) ReadUntypedMsg(rd io.Reader) (int, error) {
 	size := int(binary.BigEndian.Uint32(b.tmp[:]))
```

Comment:
If this is a `uint32` it will never be negative. Doesn't the protocol specify an `int32`?

Reply:
`binary.BigEndian` only exposes a `Uint32` method. Casting it to an `int` does the right thing here, doesn't it?